### PR TITLE
tests: Add test for the most popular images from Docker Hub.

### LINKED ISSUE #278: Add test for the most popular images from Docker Hub.
This will test kata-runtime with the most popular images from Docker Hub.

### PR DESCRIPTION
**Pull request by GabyCT**
This will test kata-runtime with the most popular images from
Docker Hub.

Fixes #278

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by devimc** [comment]
is `boltdb` image missing?

**Comment by GabyCT** [comment]
@devimc there is not a working `boltdb` image in docker hub

**Comment by egernst** [comment]
@WeiZhang555 @gnawux @bergwolf  -- can you please help augment this list to include some of the "more interesting" containers / design patterns you've run into?

File: integration/popular_docker_hub_images/popular_docker_images.bats [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+@test "start orientdb server" {
+	docker run --runtime=$RUNTIME -e ORIENTDB_ROOT_PASSWORD=mysecretword -d orientdb timeout -t 10 /orientdb/bin/server.sh
+}
+
+@test "start instance in percona container" {
+	if docker run --rm --runtime=$RUNTIME -i percona perl -e 'print "Clear Containers\n"' | grep LANG; then false; else true; fi
+}
+
+@test "run php container" {
+	docker run --rm --runtime=$RUNTIME -i php sh -c "php -r 'print(\"cc oci runtime\");'"
```
Comment (jodh-intel) [inline integration/popular_docker_hub_images/popular_docker_images.bats]: `kata-runtime` now :smile:?

File: integration/popular_docker_hub_images/popular_docker_images.bats [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+@test "run an opensuse container" {
+	docker run --rm --runtime=$RUNTIME -i opensuse sh -c "echo "testing" > test.txt | cat /test.txt | grep testing"
+}
+
+@test "start orientdb server" {
+	docker run --runtime=$RUNTIME -e ORIENTDB_ROOT_PASSWORD=mysecretword -d orientdb timeout -t 10 /orientdb/bin/server.sh
+}
+
+@test "start instance in percona container" {
+	if docker run --rm --runtime=$RUNTIME -i percona perl -e 'print "Clear Containers\n"' | grep LANG; then false; else true; fi
```
Comment (jodh-intel) [inline integration/popular_docker_hub_images/popular_docker_images.bats]: s/Clear Containers/Kata Containers/ (here and below)?

File: integration/popular_docker_hub_images/popular_docker_images.bats [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+#
+# Tests for the most popular images from docker hub.
+
+source ${BATS_TEST_DIRNAME}/../../metrics/lib/common.bash
+
+setup() {
+	clean_env
+}
+
+@test "create a directory in an aerospike container" {
```
Comment (jodh-intel) [inline integration/popular_docker_hub_images/popular_docker_images.bats]: This test is extremely useful. However what isn't immediately obvious is *why* each test was chosen (why `mkdir`? Why aerospike?)

Could you rework each `@test` slightly to include a "hint" as to why the test was included? I'd suggest using one or more "tags" in square brackets before the test description. For example, this test might become something like:

```
@test "[volume mkdir] create a directory in an aerospike container"
```

... to suggest that the test is specifically trying to test a `mkdir` on a docker volume.

This hint or tag would be a similar approach to what we do for the hypervisor config script where each configure option specifies a tag (such as `security`, `speed`, `size`). For example:

- https://github.com/kata-containers/packaging/blob/master/scripts/configure-hypervisor.sh#L307

Bonus points if you can find a way to validate the `@test` tags as we do in the hypervisor config script:

- https://github.com/kata-containers/packaging/blob/master/scripts/configure-hypervisor.sh#L21..L29

**Comment by GabyCT** [comment]
@jodh-intel changes applied

File: integration/popular_docker_hub_images/popular_docker_images.bats [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+@test "run php container" {
+	docker run --rm --runtime=$RUNTIME -i php sh -c "php -r 'print(\"kata-runtime\");'"
+}
+
+@test "check build number of a photon container" {
+	docker run --rm --runtime=$RUNTIME -i photon cat /etc/photon-release
+}
+
+@test "print a piwik container" {
+	docker run --rm --runtime=$RUNTIME -i piwik bash -c "php -r 'print(\"Clear Containers\");'"
```
Comment (jodh-intel) [inline integration/popular_docker_hub_images/popular_docker_images.bats]: Kata Containers ;)

**Comment by GabyCT** [comment]
@jodh-intel thanks changes applied

**Comment by jodh-intel** [comment]
Thanks @GabyCT.

lgtm.

**Comment by chavafg** [comment]
lgtm

**Pull request merged**